### PR TITLE
feat: add subservice 

### PR DESCRIPTION
**Pull request by Geoffrey-T**

### CODE REVIEW

File: Sources/PeerConnectionManager.swift [hunk 1]
```diff
@@ -190,7 +191,8 @@ public class PeerConnectionManager {
         self.sessionSecurityIdentity = identity
         self.sessionEncryptionPreference = encryptionPreference
 
-        self.peer = Peer(displayName: displayName)
+        self.peer = Peer(peerID: MCPeerID(displayName: displayName), status: .currentUser, info: serviceDiscoveryInfo)
+        self.subService = serviceType + (serviceDiscoveryInfo?["sessionName"] ?? "")
```
Comment (Geoffrey-T) [inline Sources/PeerConnectionManager.swift]: add const key for subservice, and rename it subservice

File: Sources/PeerConnectionManager.swift [hunk 2]
```diff
@@ -528,16 +533,30 @@ extension PeerConnectionManager {
             browserObserver.addObserver { [unowned self] event in
                 DispatchQueue.main.async {
                     switch event {
-                    case .foundPeer(let peer, _):
-                        try? self.invitePeer(peer)
+                    case .foundPeer(let peer, let discoveryInfo):
+                        print("[PEER] session info: \(self.serviceDiscoveryInfo )")
+                        print("[PEER] Found peer \(peer.displayName) with \(discoveryInfo)")
```
Comment (Geoffrey-T) [inline Sources/PeerConnectionManager.swift]: clear

File: Sources/PeerConnectionManager.swift [hunk 3]
```diff
@@ -528,16 +533,30 @@ extension PeerConnectionManager {
...
+                        let equal = NSDictionary(dictionary: discoveryInfo ?? [:])
+                            .isEqual(to: self.serviceDiscoveryInfo ?? [:])
```
Comment (Geoffrey-T) [inline Sources/PeerConnectionManager.swift]: check only subservice key ?

File: Sources/PeerConnectionManager.swift [hunk 4]
```diff
@@ -528,16 +533,30 @@ extension PeerConnectionManager {
...
                     }
                 }
             }
             advertiserObserver.addObserver { [unowned self] event in
                 DispatchQueue.main.async {
                     switch event {
-                    case .didReceiveInvitationFromPeer(peer: _, withContext: _, invitationHandler: let handler):
+                    case .didReceiveInvitationFromPeer(peer: let peer, withContext: _, invitationHandler: let handler):
+                        print("[PEER] session info: \(self.serviceDiscoveryInfo )")
+                        print("[PEER] invitation from peer \(peer.displayName)")
```
Comment (Geoffrey-T) [inline Sources/PeerConnectionManager.swift]: clear

**Review by carrarF — APPROVED** [review]

**Pull request merged**